### PR TITLE
Added Radix Sort implementation

### PR DESCRIPTION
**Pull request by rehassachdeva**
Please review
references #4 and #12

### CODE REVIEW

**Comment by aktech** [comment]
@rehassachdeva Thanks for the PR. Please resolve conflicts.

**Comment by rehassachdeva** [comment]
Hi @aktech. Please review.

File: pydsa/radix_sort.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+"""Radix Sort
+Complexity: Average Case: O(nlog(n)), where log is taken to the
+base for representing numbers. This implementation is for decimal
+system i.e. base is taken to be 10.
+"""
+
+BASE = 10
+
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: E302 expected 2 blank lines, found 1

File: pydsa/radix_sort.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+def radix_sort(a):
+    """
+    Sorts the list 'a' using Radix Sort algorithm.
+
+    >>> from pydsa import radix_sort
+    >>> a = [708, 4567, 3, 45, 911, 123, 57, 37]
+    >>> radix_sort(a)
+    [3, 37, 45, 57, 123, 708, 911, 4567]
+
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: W293 blank line contains whitespace

File: pydsa/radix_sort.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    """
+    Sorts the list 'a' using Radix Sort algorithm.
+
+    >>> from pydsa import radix_sort
+    >>> a = [708, 4567, 3, 45, 911, 123, 57, 37]
+    >>> radix_sort(a)
+    [3, 37, 45, 57, 123, 708, 911, 4567]
+    
+    """
+
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: W293 blank line contains whitespace

File: pydsa/radix_sort.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    >>> from pydsa import radix_sort
+    >>> a = [708, 4567, 3, 45, 911, 123, 57, 37]
+    >>> radix_sort(a)
+    [3, 37, 45, 57, 123, 708, 911, 4567]
+    
+    """
+ 
+    max_element = max(a)
+    n = len(a)
+
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: W293 blank line contains whitespace

File: pydsa/radix_sort.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    n = len(a)
+ 
+    # Do counting sort for every digit. Note that instead
+    # of passing digit number, exp is passed. exp is BASE^i
+    # where i is current digit number
+    exp = 1
+    while max_element//exp > 0:
+        _counting_sort(a, exp, n)
+        exp *= BASE
+
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: W293 blank line contains whitespace

File: pydsa/radix_sort.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    # of passing digit number, exp is passed. exp is BASE^i
+    # where i is current digit number
+    exp = 1
+    while max_element//exp > 0:
+        _counting_sort(a, exp, n)
+        exp *= BASE
+        
+    return a
+
+def _counting_sort(a, exp, n):
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: E302 expected 2 blank lines, found 1

File: pydsa/radix_sort.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    """Performs counting sort on list of length 'n' passed
+    as 'a' according to the digit at the 'exp' place value.
+    """
+
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: W293 blank line contains whitespace

File: pydsa/radix_sort.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    return a
+
+def _counting_sort(a, exp, n):
+    """Performs counting sort on list of length 'n' passed
+    as 'a' according to the digit at the 'exp' place value.
+    """
+ 
+    output = [0] * (n)
+    count = [0] * (BASE)
+
```
Comment (aktech) [inline pydsa/radix_sort.py]: pep8: W293 blank line contains whitespace

**Comment by aktech** [comment]
Please fix pep8 Errors.
Tip: Use a pep8 plugin to check for pep8 errors for your code editor.

**Comment by rehassachdeva** [comment]
@aktech Thanks for pointing out and for the tip. Have fixed all errors found with Flake8.

**Comment by aktech** [comment]
@rehassachdeva  Closing as new PR has been opened. #58

**Pull request closed without merging**